deepspeed/utils/timer.py

```diff
@@ -98,9 +91,9 @@ def log(self, names, normalizer=1.0, reset=True, memory_breakdown=False):
                     reset=reset) * 1000.0 / normalizer
                 string += ' | {}: {:.2f}'.format(name, elapsed_time)
 
-        # TODO: use our logging utilitied to selectively print. Useful for model
-        # parallelism because rank=0 is too restrictive.
-        print_rank_0(string)
+        # TODO: expose ranks as rank=0 is too restrictive
+        # useful for model parallelism
+        log_dist(string, ranks=[0])
```

Comment:
Can we add an optional `ranks` kwarg to `log()`? We will need more printing flexibility if using forms of model parallelism.

Reply:
Done! let me know if it looks good :)